hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -360,7 +358,8 @@ public class DataNode extends ReconfigurableBase
               FS_GETSPACEUSED_JITTER_KEY,
               FS_GETSPACEUSED_CLASSNAME));
 
-  public static final Log METRICS_LOG = LogFactory.getLog("DataNodeMetricsLog");
+  public static final org.apache.log4j.Logger METRICS_LOG =
+      org.apache.log4j.Logger.getLogger("DataNodeMetricsLog");
```

Comment:
We should avoid using log4j directly. The proper usage is to use slf4j Logger, and it will then indirectly use log4j. In fact ideally we don't even need to declare direct dependency on log4.  Certain logger usage in Hadoop forces us to use log4j APIs but in generally that's discouraged.
I don't mind making the follow up change later. This PR is already huge and take too much time to iterate.

Reply:
I agree for using slf4j directly but the problem here is that `makeMetricsLoggerAsync()` is adding async appenders and using few more APIs like `setAdditivity()` and `removeAppender()` directly from log4j.
Hence, for this PR, it would be difficult to switch to using slf4j. But yes this work is planned as a sub-task.

Reply:
at the very least, we don't want to mandate log4j on the classpath; if someone wants to use logback it is their right. so somehow this needs to be something which can be skipped if log4j is not on the CP

Reply:
Unfortunately we already have bunch of direct log4j references all over. Replacing them with slf4j as much as possible would be part of my next set of sub-tasks (parent Jira HADOOP-16206).

Reply:
is there a way the log4j refs can be kept inside MetricsLoggerTask...for example, `makeMetricsLoggerAsync` could take a log name as a string and do the lookup itself?
this would leave open a future option to move to reflection for the target log platform

Reply:
Yes that sounds good. Let me do that.